### PR TITLE
fix(platform): Reset Block Control filters on popover close

### PR DESCRIPTION
**Pull request by Swiftyos**
### Background

When you close the blocks panel, the filter does not reset. This PR fixes that

### Changes 🏗️

https://github.com/user-attachments/assets/fa7c6c29-16bf-49a3-b798-ea8395909b2d

Only for the new autogpt platform, currently in autogpt_platform/

- Create from scratch and execute an agent with at least 3 blocks
- Import an agent from file upload, and confirm it executes correctly
- Upload agent to marketplace
- Import an agent from marketplace and confirm it executes correctly
- Edit an agent from monitor, and confirm it executes correctly

### CODE REVIEW

**Comment by Swiftyos** [comment]
@Significant-Gravitas/maintainers Given we are so close to release. Please can you be extra diligent during reviews to ensure no bugs slip through

**Comment by Torantulino** [comment]
@Swiftyos thanks for this! Does this also reset the text in the search box?

**Comment by Swiftyos** [comment]
> @Swiftyos thanks for this! Does this also reset the text in the search box?

Yes it does

File: autogpt_platform/frontend/src/components/edit/control/BlocksControl.tsx [hunk 1]
```diff
@@ -53,18 +60,29 @@ export const BlocksControl: React.FC<BlocksControlProps> = ({
...
+            block.categories.some((cat) => cat.category === selectedCategory)),
+      ),
+    );
+  }, [blocks, searchQuery, selectedCategory]);
 
   return (
-    <Popover open={pinBlocksPopover ? true : undefined}>
+    <Popover
+      open={pinBlocksPopover ? true : undefined}
+      onOpenChange={(open) => {
```
Comment (majdyz) [inline autogpt_platform/frontend/src/components/edit/control/BlocksControl.tsx:80]: codestyle nit: 
```
onOpenChange={open => open || resetFilters()}
```

**Review by majdyz — APPROVED** [review]
LGTM, tested and working fine

**Pull request merged**